### PR TITLE
Selector in VolumeGroupSnapshotSource API should be optional

### PR DESCRIPTION
**Pull request by nixpanic**
**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:

When using a VolumeGroupSnapshotContent for a pre-preprovisioned VolumeGroupSnapshotSource, there is no correct value for the `selector` property. However, the original API requires the `selector` to be set, even though it is not used.

**Which issue(s) this PR fixes**:

This problem was identified in #837.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
The VolumeGroupSnapshotSource.Selector is now an optional attribute, so that a pre-provisioned VolumeGroupSnapshotContent can be specified which does not require a matching label-selector.
```

### CODE REVIEW

**Comment by nixpanic** [comment]
/cc xing-yang RaunakShah

Please have a look, combined with #837 the VolumeGroupSnapshot feature should be pretty much ready for users to try out.

File: client/apis/volumegroupsnapshot/v1alpha1/types.go [hunk 1]
```diff
@@ -52,8 +52,8 @@ type VolumeGroupSnapshotSource struct {
 	// is created, the existing group snapshots won't be modified.
 	// Once a VolumeGroupSnapshotContent is created and the sidecar starts to process
 	// it, the volume list will not change with retries.
-	// Required.
-	Selector metav1.LabelSelector `json:"selector" protobuf:"bytes,1,opt,name=selector"`
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector" protobuf:"bytes,1,opt,name=selector"`
```
Comment (xing-yang) [inline client/apis/volumegroupsnapshot/v1alpha1/types.go:56]: It should have "omitempty".

Comment (nixpanic) [reply client/apis/volumegroupsnapshot/v1alpha1/types.go]: That's been done now too.

**Review by RaunakShah — APPROVED** [review]
Thanks for catching this!

**Pull request merged**